src/Analyzer.Reports/SarifReportWriter.cs

```diff
@@ -148,7 +150,7 @@ private void ExtractResult(IEvaluation rootEvaluation, IEvaluation evaluation, s
                                     Uri = new Uri
                                     (
                                         UriHelper.MakeValidUri(filePath),
-                                        UriKind.Relative
+                                        UriKind.RelativeOrAbsolute
                                     ),
                                     UriBaseId = UriBaseIdString,
```

Comment:
Shouldn't the `UriBaseId` not be set if `filePath` ends up being a full path, not having a root equal to `uriBaseIdString`, as a result of line 60?  It's possible this could happen when processing nested templates - the nested template could be outside the directory scanned, or possibly even a URL in the future.

Reply:
Updated code to not set UriBaseId if templates path does not belong to the root path.